plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/HiveMetastoreBackedDeltaLakeMetastore.java

```diff
@@ -118,6 +120,10 @@ public Optional<Table> getTable(String databaseName, String tableName)
     {
         Optional<Table> candidate = delegate.getTable(databaseName, tableName);
         candidate.ifPresent(table -> {
+            if (isPrestoView(table) && isHiveOrPrestoView(table)) {
+                // this is a Hive view, hence not a table
+                throw new TableNotFoundException(new SchemaTableName(databaseName, tableName));
```

Comment:
Should we throw, or return `empty`?

Reply:
I took this behaviour  from https://github.com/trinodb/trino/blob/master/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueIcebergTableOperations.java#L76-L79

We probably should have a consistent way of handling this kind of situations across hive/iceberg/delta.

`getTable(tableName)` should probably point out that the `tableName` points to a view - so `HiveViewNotSupportedException` / `TableNotFoundException` are (not perfect, but) good candidates

Reply:
`HiveViewNotSupportedException` is suitable for a hive view.

for `isPrestoView` we could throw a similar exception (just for now, until we add support for Trino/Presto views in Delta)
would that work @findinpath ?

Reply:
What about throwing `NotADeltaLakeTableException` exception?
This exception is already used in case of dealing with a Hive table.
Wouldn't it be appropriate to do the same in case of Hive views?

Reply:
`NotADeltaLakeTableException` is awesome & applicable to both Hive and Trino views, thanks @findinpath for the suggstion